firmware/src/services/iot/cloud/cloud_service.c

```diff
@@ -0,0 +1,628 @@
+/*
+    \file   cloud_service.c
+
+    \brief  Cloud Service Abstraction Layer
+
+    (c) 2018 Microchip Technology Inc. and its subsidiaries.
+
+    Subject to your compliance with these terms, you may use Microchip software and any
+    derivatives exclusively with Microchip products. It is your responsibility to comply with third party
+    license terms applicable to your use of third party software (including open source software) that
+    may accompany Microchip software.
+
+    THIS SOFTWARE IS SUPPLIED BY MICROCHIP "AS IS". NO WARRANTIES, WHETHER
+    EXPRESS, IMPLIED OR STATUTORY, APPLY TO THIS SOFTWARE, INCLUDING ANY
+    IMPLIED WARRANTIES OF NON-INFRINGEMENT, MERCHANTABILITY, AND FITNESS
+    FOR A PARTICULAR PURPOSE.
+
+    IN NO EVENT WILL MICROCHIP BE LIABLE FOR ANY INDIRECT, SPECIAL, PUNITIVE,
+    INCIDENTAL OR CONSEQUENTIAL LOSS, DAMAGE, COST OR EXPENSE OF ANY KIND
+    WHATSOEVER RELATED TO THE SOFTWARE, HOWEVER CAUSED, EVEN IF MICROCHIP
+    HAS BEEN ADVISED OF THE POSSIBILITY OR THE DAMAGES ARE FORESEEABLE. TO
+    THE FULLEST EXTENT ALLOWED BY LAW, MICROCHIP'S TOTAL LIABILITY ON ALL
+    CLAIMS IN ANY WAY RELATED TO THIS SOFTWARE WILL NOT EXCEED THE AMOUNT
+    OF FEES, IF ANY, THAT YOU HAVE PAID DIRECTLY TO MICROCHIP FOR THIS
+    SOFTWARE.
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include "../../../iot_config/cloud_config.h"
+#include "cloud_service.h"
+#include "../../../iot_config/IoT_Sensor_Node_config.h"
+#include "crypto_client/crypto_client.h"
+#include "crypto_client/cryptoauthlib_main.h"
+#include "debug_print.h"
+#include "m2m_wifi.h"
+#include "../../../app.h"
+#include "bsd_adapter/bsdWINC.h"
+#include "socket.h"
+#include "../cloud/mqtt_packetPopulation/mqtt_packetPopulate.h"
+#include "../../../mqtt/mqtt_core/mqtt_core.h"
+#include "wifi_service.h"
+#include "../../../credentials_storage/credentials_storage.h"
+#include "../../../mqtt/mqtt_packetTransfer_interface.h"
+#include "definitions.h"
+#include "iot_config/mqtt_config.h"
+#include "led.h"
+#include "../../../config/SAMD21_WG_IOT/driver/winc/include/drv/driver/m2m_ssl.h"
+
+#define UNIX_OFFSET 946684800
+
+static bool cloudInitialized = false;
+static bool waitingForMQTT   = false;
+
+pf_MQTT_CLIENT*   pf_mqtt_client;
+char*             mqtt_host;
+volatile uint32_t mqttHostIP;
+volatile uint32_t dnsRetryCount = 0;
+char              mqttSubscribeTopic[TOPIC_SIZE];
+
+static int8_t  connectMQTTSocket(void);
+static void    connectMQTT();
+static uint8_t reInit(void);
+
+// bool isResetting      = false;
+bool sendSubscribe = true;
+#define CLOUD_TASK_INTERVAL      500L
+#define CLOUD_MQTT_TIMEOUT_COUNT 30000L   // 10 seconds max allowed to establish a connection
+#define MQTT_CONN_AGE_TIMEOUT    3600L    // 3600 seconds = 60minutes
+#define CLOUD_RESET_TIMEOUT      2000L    // 2 seconds
+#define DNS_RETRY_COUNT          10000L / CLOUD_TASK_INTERVAL
+#define WIFI_CONNECT_TIMEOUT     5000L   // 5 seconds

```

Comment:
all of these would benefit from units on the end of the macro names: double check suggestions are right
```suggestion
#define CLOUD_TASK_INTERVAL_MS      500L
#define CLOUD_MQTT_TIMEOUT_COUNT_MS 30000L   // 10 seconds max allowed to establish a connection
#define MQTT_CONN_AGE_TIMEOUT_SEC   3600L    // 3600 seconds = 60minutes
#define CLOUD_RESET_TIMEOUT_MS      2000L    // 2 seconds
#define DNS_RETRY_COUNT_MS          10000L / CLOUD_TASK_INTERVAL
#define WIFI_CONNECT_TIMEOUT_MS     5000L   // 5 seconds

```

Reply:
updated